### DISCUSSION TITLE
deepgram websocket for text to speech

### DISCUSSION THREAD

**Discussion by codexmuneer** [deepgram/community#666] [General help] [answered]
Is there any Deepgram Websocket available for new text to speech model aura-helios-en?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by codexmuneer** [reply]
I was utilizing speech-to-text through the Deepgram WebSocket URL "wss://api.deepgram.com/v1/listen?encoding=ENCODING_VALUE&sample_rate=SAMPLE_RATE_VALUE," and text-to-speech via Elevenlabs for an AI voicebot. Now, I'm interested in experimenting with Deepgram's new text-to-speech feature instead of Elevenlabs. However, I couldn't find any documentation regarding text-to-speech streaming via a WebSocket connection.
 This code snippet is given for streaming audio but is there a way to do it using websocket connection.
 import requests

```python
DEEPGRAM_URL = "https://api.deepgram.com/v1/speak?model=aura-asteria-en"
DEEPGRAM_API_KEY = "DEEPGRAM_API_KEY"

payload = {
    "text": "Hello, how can I help you today? My name is Emily and I'm very glad to meet you. What do you think of this new text-to-speech API?"
}

headers = {
    "Authorization": f"Token {DEEPGRAM_API_KEY}",
    "Content-Type": "application/json"
}

audio_file_path = "output.mp3"  # Path to save the audio file

with open(audio_file_path, 'wb') as file_stream:
    response = requests.post(DEEPGRAM_URL, headers=headers, json=payload, stream=True)
    for chunk in response.iter_content(chunk_size=1024):
        if chunk:
            file_stream.write(chunk) # Write each chunk of audio data to the file

print("Audio download complete")
```

**Comment by jpvajda** [comment, marked as answer]
@codexmuneer  We currently don't have a websocket API for Aura Text to Speech, but it's high on our list of priorities to offer this in the near future. We do provide output streaming which might be interesting to you depending on your use case.

With Aura, the audio file is streamed back to you, so you may begin playback as soon as the first byte arrives. Read the guide [Streaming Audio Outputs](https://developers.deepgram.com/docs/streaming-the-audio-output) to learn how to begin playing the stream immediately versus waiting for the entire file to arrive.
